### PR TITLE
InventoryBundle module

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
Created new module for bundle product specific logic

### Fixed Issues 
1. magento-engcom/msi#380: Bundle product get "Out of stock" status when created

### Manual testing scenarios
1. Simple product created with Qantity 100
2. Create Bundle product with your Simple product as "Budle Items"
3. Bundle product should be "In stock"

### CODE REVIEW

File: app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php [hunk 1]
```diff
@@ -73,6 +73,9 @@ public function execute(EventObserver $observer)
     {
         /** @var ProductInterface $product */
         $product = $observer->getEvent()->getProduct();
+        if ($product->getTypeId() == 'bundle') {
```
Comment (naydav) [inline app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php]: Need to introduce new module `InventoryBundle`
And disable this observer in new module
Test also need to move to new module

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php (hunk 1 above)
Comment (naydav) [reply app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php]: Also need to use this constant
`\Magento\Bundle\Model\Product\Type::TYPE_CODE`

Comment (naydav) [reply app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php]: And looks like we need to disable data provider plugin in Inventory module

Comment (VitaliyBoyko) [reply app/code/Magento/InventoryCatalog/Observer/ProcessSourceItemsObserver.php]: Valeriy, I don't get which plugin in Inventory module should be disabled. There is no any plugins.

File: app/code/Magento/InventoryBundle/Plugin/InventoryCatalog/Observer/DisableProcessSourceItemsObserverPlugin.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+use Magento\Bundle\Model\Product\Type;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\InventoryCatalog\Observer\ProcessSourceItemsObserver;
+use Magento\Framework\Event\Observer;
+
+/**
+ * Class disable inventory for bundle products
+ */
+class DisableProcessSourceItemsObserverPlugin
```
Comment (naydav) [inline app/code/Magento/InventoryBundle/Plugin/InventoryCatalog/Observer/DisableProcessSourceItemsObserverPlugin.php]: You need to disable observer via events configuration

We don't allow use plugins for plugins, observers, data objects
Generally, we think about mark these classes as `final` for avoiding pluginization

Comment (VitaliyBoyko) [reply app/code/Magento/InventoryBundle/Plugin/InventoryCatalog/Observer/DisableProcessSourceItemsObserverPlugin.php]: Ok, get it.

**Comment by naydav** [comment]
`Magento\InventoryBundle\Test\Integration\Catalog\Controller\Adminhtml\ProductControllerSaveActionTest` will be removed in scope of https://github.com/magento-engcom/msi/issues/370

**Pull request merged**